streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -665,6 +667,50 @@ public void shouldAddTimestampExtractorWithPatternPerSource() throws Exception {
     }
 
     @Test
+    public void shouldSortProcessorNodesCorrectly() throws Exception {
+        builder.addSource(null, "source1", null, null, null, "topic1");
+        builder.addSource(null, "source2", null, null, null, "topic2");
+        builder.addProcessor("processor1", new MockProcessorSupplier(), "source1");
+        builder.addProcessor("processor2", new MockProcessorSupplier(), "source1", "source2");
+        builder.addProcessor("processor3", new MockProcessorSupplier(), "processor2");
+        builder.addSink("sink1", "topic2", null, null, null, "processor1", "processor3");
+
+        assertEquals(1, builder.describe().subtopologies().size());
+
+        final Iterator<TopologyDescription.Node> iterator = ((InternalTopologyBuilder.Subtopology) builder.describe().subtopologies().iterator().next()).nodesInOrder();
+
+        assertTrue(iterator.hasNext());
+        InternalTopologyBuilder.AbstractNode node = (InternalTopologyBuilder.AbstractNode) iterator.next();
+        assertTrue(node.name.equals("source1"));
```

Comment:
Can we use `assertThat(node.name, equalTo("source1")` or `assertEquals("source1", node.name)` instead of `assertTrue` for assertions like this?
Elsewhere in this test, too

Reply:
`assertEquals` does not work perfectly for String types, I will try `assertThat` then.